### PR TITLE
Fix Issue 16181 - Overloading doesn't consider default arguments with…

### PR DESCRIPTION
**Pull request by Luhrel**
… regards to ambiguity

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by kubo39** [comment]
should it be deprecation?

File: src/dmd/dtemplate.d [hunk 1]
```diff
@@ -2717,8 +2717,25 @@ void functionResolve(ref MatchAccumulator m, Dsymbol dstart, Loc loc, Scope* sc,
...
+                    TypeFunction tf_ = f.type.isTypeFunction();
+                    size_t nparams = tf_.parameterList.length;
+                    if (nparams > nargs && tf_.parameterList[nargs].defaultArg)
+                    {
+                        bypass = true;
+                    }
+                }
+
+                if (!bypass)
+                {
```
Comment (kubo39) [inline src/dmd/dtemplate.d:2735]: NIT: c1 and c2 can be moved here.

**Comment by WalterBright** [comment]
It is not a bug and the compiler is behaving as designed.

**Comment by WalterBright** [comment]
Sorry.

**Pull request closed without merging**